manifests/profile/proxied_app.pp

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2019 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# Declaration of a service to be made available via reverse proxy. This
+# should be exported by an app and then collected by the web server role or
+# profile.
+
+# There are many keys in hiera for each instance, including some new ones
+# for this profile:
+#   public_hostname: hostname users access
+#   url_root: root url segment/suffix on the public hostname
+#   protocol: which protocol the app serves, http or https
+#   hostname: the hostname for the app as reachable by the web server
+#   port: port the app listens on
+#   sendfile_path: if set, make this path available for X-Sendfile
+
+class nebula::profile::proxied_app(
+  String  $public_hostname,
+  String  $url_root,
+  String  $protocol, # internal protocol, not front-end
+  String  $hostname,
+  Integer $port,
+  String  $static_path,
+  String  $sendfile_path,     # If set, XSendFile will be enabled here
+){
+  # These are straightforward translations to maintain parity with the
+  # ansible template. All of these names and semantics should be revisited
+  # and likely decomposed into resources/fragments that can be assembled
+  # by the web host after being exported or virtualized wherever needed.
+  $apache_app_hostname  = $hostname
+  $apache_app_name      = $title
+  $apache_cosign_factor = "UMICH.EDU"
+  $apache_domain        = $public_hostname
+  $apache_port          = $port
+  $apache_ssl_crt       = "${public_hostname}.crt"
```

Comment:
The ansible setup takes the name of the ssl cert as a parameter, and I think we will probably want to do that too.

Reply:
Thanks. These suggestions are on the branch now.